include/nbl/builtin/glsl/ext/RadixSort/radix_sort.glsl

```diff
@@ -0,0 +1,69 @@
+#ifndef _NBL_GLSL_EXT_RADIXSORT_INCLUDED_
+#define _NBL_GLSL_EXT_RADIXSORT_INCLUDED_
+
+#define _NBL_GLSL_SCRATCH_SHARED_SIZE_DEFINED_ (_NBL_GLSL_WORKGROUP_SIZE_ * 2 * 4)
+shared uint scratch_shared[_NBL_GLSL_SCRATCH_SHARED_SIZE_DEFINED_];
+#define _NBL_GLSL_SCRATCH_SHARED_DEFINED_ scratch_shared
+
+#include <nbl/builtin/glsl/workgroup/ballot.glsl>
+
+#include "nbl/builtin/glsl/ext/RadixSort/parameters.glsl"
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_
+#error "You need to define `nbl_glsl_ext_RadixSort_getParameters` and mark `_NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_`!"
+#endif
+
+uint nbl_glsl_ext_RadixSort_extractDigit(in uint key)
+{
+	return (key >> nbl_glsl_ext_RadixSort_Parameters_t_getShift()) & 0xf;
```

Comment:
the mask should be derived as `uint(NUM_BUCKETS-1)` not a constant literal like `0xf`